### PR TITLE
api: specialize waitForEvent

### PR DESCRIPTION
**Pull request by pavelfeldman**

### CODE REVIEW

File: src/Playwright.Tests/PageBasicTests.cs [hunk 1]
```diff
@@ -204,8 +204,11 @@ public async Task ShouldWorkWithPageClose()
         [Fact(Timeout = TestConstants.DefaultTestTimeout)]
         public async Task ShouldFireLoadWhenExpected()
         {
+            var loadEvent = new TaskCompletionSource();
```
Comment (kblok) [inline src/Playwright.Tests/PageBasicTests.cs:207]: Let's make the tests .NET 3.1 friendly. If we don't want to be 3.1 friendly let's remove the build step.

```suggestion
            var loadEvent = new TaskCompletionSource<bool>();
```

File: src/Playwright.Tests/PageBasicTests.cs [hunk 2]
```diff
@@ -204,8 +204,11 @@ public async Task ShouldWorkWithPageClose()
...
+            Page.Load += (_, _) => loadEvent.TrySetResult();
```
Comment (kblok) [inline src/Playwright.Tests/PageBasicTests.cs:208]: Let's make the tests .NET 3.1 friendly. If we don't want to be 3.1 friendly let's remove the build step.

```suggestion
            Page.Load += (_, _) => loadEvent.TrySetResult(true);
```

File: src/Playwright/API/Generated/Types/BrowserContextCookiesResult.cs [hunk 3]
```diff
@@ -37,7 +37,7 @@
 namespace Microsoft.Playwright
 {
     /// <summary>
-    /// Result of calling <see cref="IBrowserContext.CookiesAsync"/>.
+    /// Result of calling <see cref="IBrowserContext.Cookies"/>.
```
Comment (dgozman) [inline src/Playwright/API/Generated/Types/BrowserContextCookiesResult.cs:40]: Why this change?

File: src/Playwright/API/Generated/Types/ElementHandleBoundingBoxResult.cs [hunk 4]
```diff
@@ -37,7 +37,7 @@
 namespace Microsoft.Playwright
 {
     /// <summary>
-    /// Result of calling <see cref="IElementHandle.BoundingBoxAsync"/>.
+    /// Result of calling <see cref="IElementHandle.BoundingBox"/>.
```
Comment (dgozman) [inline src/Playwright/API/Generated/Types/ElementHandleBoundingBoxResult.cs:40]: Why this change?

File: src/Playwright.Tests/PageBasicTests.cs (hunk 1 above)
Comment (kblok) [reply src/Playwright.Tests/PageBasicTests.cs:207]: We get the same in other places. Let's make a decision and apply it across the board.

**Review by dgozman — APPROVED** [review]

File: src/Playwright.Tests/PageEventPopupTests.cs [hunk 5]
```diff
@@ -79,27 +79,15 @@ public async Task ShouldEmitForImmediatelyClosedPopupsWithLocation()
         [Fact(Timeout = TestConstants.DefaultTestTimeout)]
         public async Task ShouldBeAbleToCaptureAlert()
```
Comment (kblok) [inline src/Playwright.Tests/PageEventPopupTests.cs:80]:
```suggestion
        public void ShouldBeAbleToCaptureAlert()
```

File: src/Playwright/API/Generated/Types/BrowserContextCookiesResult.cs (hunk 3 above)
Comment (pavelfeldman) [reply src/Playwright/API/Generated/Types/BrowserContextCookiesResult.cs:40]: Ooh, this is not right!

File: src/Playwright.Tests/FrameHierarchyTests.cs [hunk 6]
```diff
@@ -191,8 +191,11 @@ public async Task ShouldReportDifferentFrameInstanceWhenFrameReAttaches()
             }");
             Assert.True(frame1.IsDetached);
 
+            var frameEvent = new TaskCompletionSource<IFrame>();
+            Page.FrameNavigated += (_, frame) => frameEvent.TrySetResult(frame);
```
Comment (kblok) [inline src/Playwright.Tests/FrameHierarchyTests.cs:195]: Comparing the two changes in this file. How would a customer know when to use `WaitForNavigationAsync` and when they would need to implement that mannualy?
